graph/src/util/cache_weight.rs

```diff
@@ -30,6 +31,12 @@ impl<T: CacheWeight> CacheWeight for Vec<T> {
     }
 }
 
+impl<T: CacheWeight> CacheWeight for Arc<T> {
```

Comment:
I don't think we should have this blanket impl, `Arc` is case where the ownership and therefore the weight of the value is context dependent. I'd suggest removing this impl as it is not really necessary and the cache weight for an `Arc` should be justified in each context.

Reply:
Agreed, changed